wp-includes/class-wp-https-ui.php

```diff
@@ -0,0 +1,220 @@
+				echo wp_kses_post( sprintf(
+					/* Translators: %s: a link for more details */
+					esc_html__( 'HTTPS is essential to securing your WordPress site, we strongly suggest upgrading to HTTPS on your site. %s', 'pwa' ),
+					$https_more_details
+				) );
+				?>
+			</p>
+			<p style="margin-top: 20px;"><strong><?php esc_html_e( 'HTTPS Upgrade', 'pwa' ); ?></strong></p>
+			<p>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_HTTPS_OPTION ); ?>" type="radio" <?php checked( $upgrade_https_value, self::OPTION_SELECTED_VALUE ); ?> value="<?php echo esc_attr( self::OPTION_SELECTED_VALUE ); ?>"><?php esc_html_e( 'Yes', 'pwa' ); ?></label>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_HTTPS_OPTION ); ?>" type="radio" <?php checked( self::OPTION_SELECTED_VALUE !== $upgrade_https_value ); ?> value="0" ><?php esc_html_e( 'No', 'pwa' ); ?></label>
+			</p>
+			<p class="description"><?php esc_html_e( 'Your site appears to support HTTPS', 'pwa' ); ?></p>
+
+			<p style="margin-top: 20px;"><strong><?php esc_html_e( 'Upgrade Insecure URLs', 'pwa' ); ?></strong></p>
+			<p>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_INSECURE_CONTENT_OPTION ); ?>" type="radio" <?php checked( $upgrade_insecure_content, self::OPTION_SELECTED_VALUE ); ?> value="<?php echo esc_attr( self::OPTION_SELECTED_VALUE ); ?>"><?php esc_html_e( 'Yes', 'pwa' ); ?></label>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_INSECURE_CONTENT_OPTION ); ?>" type="radio" <?php checked( self::OPTION_SELECTED_VALUE !== $upgrade_insecure_content ); ?> value="0" ><?php esc_html_e( 'No', 'pwa' ); ?></label>
+			</p>
+			<p class="description">
+				<?php echo wp_kses_post( sprintf( $insecure_content_description, $insecure_content_more_details ) ); ?>
+			<p>
+			<?php
+		else :
+			/* Translators: %s: HTTPS more details link */
+			echo wp_kses_post( sprintf( __( 'Your site doesn\'t look to support HTTPS. %s', 'pwa' ), $https_more_details ) );
+		endif;
+	}
+
+	/**
+	 * Whether the options indicate that the site is currently using HTTPS.
+	 *
+	 * @return boolean
+	 */
+	public function is_currently_https() {
+		$urls = array( get_option( 'home' ), get_option( 'siteurl' ) );
+		foreach ( $urls as $url ) {
+			if ( 0 !== strpos( $url, self::HTTPS_PROTOCOL ) ) {
+				return false;
+			}
+		}
+		return true;
+
+	}
+
+	/**
+	 * Conditionally filters the 'siteurl' and 'home' values from wp-config and options.
+	 */
+	public function filter_site_url_and_home() {
+		if ( self::OPTION_SELECTED_VALUE === get_option( self::UPGRADE_HTTPS_OPTION ) ) {
+			add_filter( 'option_home', array( $this, 'convert_to_https' ), 11 );
+			add_filter( 'option_siteurl', array( $this, 'convert_to_https' ), 11 );
+		}
+	}
+
+	/**
+	 * Converts a URL from HTTP to HTTPS
+	 *
+	 * @param string $url The URL to convert.
+	 * @return string $url The converted URL.
+	 */
+	public function convert_to_https( $url ) {
+		return str_replace( 'http://', self::HTTPS_PROTOCOL, $url );
```

Comment:
This is not as robust as it could be. What if I have a URL like: `https://example.com/redirect/url/http://example.org/`. This will not then work as expected. So it would be better to make sure only the beginning is converted, for example:

```php
return preg_replace( '#^http(?=://)#', 'https', $url );
```

Reply:
Thanks, [https://github.com/xwp/pwa-wp/pull/36/commits/330ae829561bcbf99f1741f0e7437c5dcb3b3b4d](https://github.com/xwp/pwa-wp/pull/36/commits/330ae829561bcbf99f1741f0e7437c5dcb3b3b4d) uses your `preg_replace()` snippet above, and adds a test assertion.